Data set from COVID pandemic taken from kaggle: https://www.kaggle.com/datasets/ghost5612/novel-covid-19-dataset/data

Sno – Serial number of the record
ObservationDate – Date of the observation (MM/DD/YYYY)
Province/State – Province or state of the observation (may be missing for some entries)
Country/Region – Country of the observation
Last Update – Timestamp (UTC) when the record was last updated (not standardized, requires cleaning before use)
Confirmed – Cumulative number of confirmed cases on that date
Deaths – Cumulative number of deaths on that date
Recovered – Cumulative number of recoveries on that date

There are multiple workbooks so we will make multiple DataFrames

In [10]:
import pandas as pd

'''
Abbreviations
ts - time series
us - United States
'''

ts_confirmed_us = pd.read_csv(r"C:\Users\tombo\OneDrive\Documents\Python Projects\Data\time_series_covid_19_confirmed_US.csv")
ts_confirmed = pd.read_csv(r"C:\Users\tombo\OneDrive\Documents\Python Projects\Data\time_series_covid_19_confirmed.csv")
ts_deaths_us = pd.read_csv(r"C:\Users\tombo\OneDrive\Documents\Python Projects\Data\time_series_covid_19_deaths_US.csv")
ts_deaths = pd.read_csv(r"C:\Users\tombo\OneDrive\Documents\Python Projects\Data\time_series_covid_19_deaths.csv")
ts_recovered = pd.read_csv(r"C:\Users\tombo\OneDrive\Documents\Python Projects\Data\time_series_covid_19_recovered.csv")



Now to explore our DataFrames to understand their columns and records.

In [11]:
ts_confirmed_us.head(10)

UID iso2 iso3  code3    FIPS    Admin2 Province_State Country_Region  \
0  84001001   US  USA    840  1001.0   Autauga        Alabama             US   
1  84001003   US  USA    840  1003.0   Baldwin        Alabama             US   
2  84001005   US  USA    840  1005.0   Barbour        Alabama             US   
3  84001007   US  USA    840  1007.0      Bibb        Alabama             US   
4  84001009   US  USA    840  1009.0    Blount        Alabama             US   
5  84001011   US  USA    840  1011.0   Bullock        Alabama             US   
6  84001013   US  USA    840  1013.0    Butler        Alabama             US   
7  84001015   US  USA    840  1015.0   Calhoun        Alabama             US   
8  84001017   US  USA    840  1017.0  Chambers        Alabama             US   
9  84001019   US  USA    840  1019.0  Cherokee        Alabama             US   

         Lat      Long_  ... 5/20/21  5/21/21  5/22/21  5/23/21  5/24/21  \
0  32.539527 -86.644082  ...    7049     7106     7113     7118     7118   
1  30.727750 -87.722071  ...   21489    21511    21535    21546    21554   
2  31.868263 -85.387129  ...    2326     2327     2328     2328     2328   
3  32.996421 -87.125115  ...    2656     2657     2656     2658     2659   
4  33.982109 -86.567906  ...    6816     6826     6829     6832     6832   
5  32.100305 -85.712655  ...    1233     1233     1233     1233     1233   
6  31.753001 -86.680575  ...    2195     2199     2201     2202     2202   
7  33.774837 -85.826304  ...   14590    14588    14595    14606    14608   
8  32.913601 -85.390727  ...    3622     3634     3636     3640     3641   
9  34.178060 -85.606390  ...    1852     1852     1854     1858     1858   

   5/25/21  5/26/21  5/27/21  5/28/21  5/29/21  
0     7126     7135     7141     7142     7142  
1    21578    21593    21606    21620    21620  
2     2331     2331     2333     2334     2334  
3     2660     2662     2666     2664     2664  
4     6847     6856     6862     6864     6864  
5     1233     1233     1233     1233     1233  
6     2210     2216     2215     2219     2219  
7    14606    14614    14621    14622    14622  
8     3653     3654     3662     3665     3665  
9     1859     1859     1861     1862     1862  

[10 rows x 505 columns]

Each record has a unique ID. All the records are based in the US, now we could group the records by province to compare provinces against each other.

In [16]:
ts_us_byprovince = ts_confirmed_us.groupby(by="Province_State").mean(numeric_only=True)
ts_us_byprovince

UID  code3          FIPS        Lat  \
Province_State                                                           
Alabama                   8.400350e+07  840.0   3499.869565  31.931113   
Alaska                    8.400717e+07  840.0   7165.848485  56.628273   
American Samoa            1.600000e+01   16.0     60.000000 -14.271000   
Arizona                   8.401354e+07  840.0  13542.117647  29.714033   
Arkansas                  8.400715e+07  840.0   7151.103896  34.005087   
California                8.400869e+07  840.0   8689.600000  36.582496   
Colorado                  8.401039e+07  840.0  10393.924242  37.755612   
Connecticut               8.402421e+07  840.0  24208.200000  33.290944   
Delaware                  8.404001e+07  840.0  40005.800000  23.465566   
Diamond Princess          8.408889e+07  840.0  88888.000000   0.000000   
District of Columbia      8.406034e+07  840.0  60341.000000  12.968059   
Florida                   8.401418e+07  840.0  14182.231884  28.101892   
Georgia                   8.401405e+07  840.0  14054.055901  32.400973   
Grand Princess            8.410000e+07  840.0  99999.000000   0.000000   
Guam                      3.160000e+02  316.0     66.000000  13.444300   
Hawaii                    8.403501e+07  840.0  35007.857143  15.020202   
Idaho                     8.401904e+07  840.0  19042.782609  42.338906   
Illinois                  8.401841e+07  840.0  18408.057692  39.077394   
Indiana                   8.401952e+07  840.0  19515.957447  38.989750   
Iowa                      8.402040e+07  840.0  20404.346535  41.197635   
Kansas                    8.402132e+07  840.0  21318.364486  37.766026   
Kentucky                  8.402217e+07  840.0  22167.557377  37.005387   
Louisiana                 8.402397e+07  840.0  23971.818182  30.150233   
Maine                     8.402991e+07  840.0  29905.666667  39.722390   
Maryland                  8.402874e+07  840.0  28735.653846  36.013773   
Massachusetts             8.403472e+07  840.0  32515.375000  37.117303   
Michigan                  8.402845e+07  840.0  27469.894118  42.018626   
Minnesota                 8.402839e+07  840.0  28389.022472  44.528372   
Mississippi               8.402944e+07  840.0  29437.857143  32.011947   
Missouri                  8.403041e+07  840.0  30073.461538  37.804955   
Montana                   8.403195e+07  840.0  31951.655172  45.354874   
Nebraska                  8.403223e+07  840.0  32228.536842  40.451862   
Nevada                    8.403762e+07  840.0  37623.052632  35.123787   
New Hampshire             8.404168e+07  840.0  41680.500000  36.232477   
New Jersey                8.403846e+07  840.0  38456.913043  36.783272   
New Mexico                8.403789e+07  840.0  37887.571429  32.575110   
New York                  8.403759e+07  840.0  37592.437500  41.214647   
North Carolina            8.403804e+07  840.0  38039.941176  34.925696   
North Dakota              8.403976e+07  840.0  39761.545455  45.670106   
Northern Mariana Islands  5.800000e+02  580.0     69.000000  15.097900   
Ohio                      8.404011e+07  840.0  40109.133333  39.414877   
Oklahoma                  8.404122e+07  840.0  41215.303797  34.603132   
Oregon                    8.404335e+07  840.0  43352.052632  42.150695   
Pennsylvania              8.404331e+07  840.0  43312.652174  39.634946   
Puerto Rico               6.307210e+07  630.0  72098.375000  17.783793   
Rhode Island              8.405573e+07  840.0  55730.428571  29.753673   
South Carolina            8.404671e+07  840.0  46712.625000  32.542577   
South Dakota              8.404721e+07  840.0  47214.367647  42.970557   
Tennessee                 8.404788e+07  840.0  47877.515464  35.145203   
Texas                     8.404854e+07  840.0  48541.453125  31.413295   
Utah                      8.405438e+07  840.0  51352.870968  37.528405   
Vermont                   8.405439e+07  840.0  54393.500000  38.639559   
Virgin Islands            8.500000e+02 

Get column values

In [13]:
ts_us_byprovince.columns.values

array(['UID', 'code3', 'FIPS', 'Lat', 'Long_', '1/22/20', '1/23/20',
       '1/24/20', '1/25/20', '1/26/20', '1/27/20', '1/28/20', '1/29/20',
       '1/30/20', '1/31/20', '2/1/20', '2/2/20', '2/3/20', '2/4/20',
       '2/5/20', '2/6/20', '2/7/20', '2/8/20', '2/9/20', '2/10/20',
       '2/11/20', '2/12/20', '2/13/20', '2/14/20', '2/15/20', '2/16/20',
       '2/17/20', '2/18/20', '2/19/20', '2/20/20', '2/21/20', '2/22/20',
       '2/23/20', '2/24/20', '2/25/20', '2/26/20', '2/27/20', '2/28/20',
       '2/29/20', '3/1/20', '3/2/20', '3/3/20', '3/4/20', '3/5/20',
       '3/6/20', '3/7/20', '3/8/20', '3/9/20', '3/10/20', '3/11/20',
       '3/12/20', '3/13/20', '3/14/20', '3/15/20', '3/16/20', '3/17/20',
       '3/18/20', '3/19/20', '3/20/20', '3/21/20', '3/22/20', '3/23/20',
       '3/24/20', '3/25/20', '3/26/20', '3/27/20', '3/28/20', '3/29/20',
       '3/30/20', '3/31/20', '4/1/20', '4/2/20', '4/3/20', '4/4/20',
       '4/5/20', '4/6/20', '4/7/20', '4/8/20', '4/9/20', '4/10/20',
       '4

The data is cumulative so we can get an idea of the last known count by using the last date which is 5/29/21. Now we will find the province with the most cases.
First to select the series.

In [14]:
final_date = ts_us_byprovince['5/29/21']
final_date.values

array([7.87543478e+03, 2.12751515e+03, 0.00000000e+00, 5.17921176e+04,
       4.43233766e+03, 6.31452167e+04, 8.21825758e+03, 3.47341000e+04,
       2.17540000e+04, 4.90000000e+01, 1.62993333e+04, 3.36350435e+04,
       6.97728571e+03, 1.03000000e+02, 8.15600000e+03, 5.16057143e+03,
       4.17417391e+03, 1.32786346e+04, 7.90785106e+03, 3.67654455e+03,
       2.94701869e+03, 3.75635246e+03, 7.13159091e+03, 3.75877778e+03,
       1.76741538e+04, 4.15754706e+04, 1.14016207e+04, 6.75398876e+03,
       3.77865476e+03, 5.13622881e+03, 1.92732759e+03, 2.35056842e+03,
       1.70393684e+04, 8.22491667e+03, 4.41690870e+04, 5.79488571e+03,
       3.28289844e+04, 9.81523529e+03, 1.99845455e+03, 1.83000000e+02,
       1.22395222e+04, 5.73135443e+03, 5.28960526e+03, 1.74335072e+04,
       1.73223750e+03, 2.16775714e+04, 1.23523333e+04, 1.82591176e+03,
       8.89073196e+03, 1.15346133e+04, 1.09689459e+04, 1.51250000e+03,
       3.44200000e+03, 5.00122222e+03, 1.06304634e+04, 2.82959649e+03,
      

Now to get the index of the highest number in this series and use that index to reference the province series.

In [20]:
idx = final_date.idxmax()
max_province = ts_us_byprovince.loc[idx]
max_province

UID        8.400869e+07
code3      8.400000e+02
FIPS       8.689600e+03
Lat        3.658250e+01
Long_     -1.167043e+02
               ...     
5/25/21    6.299997e+04
5/26/21    6.301867e+04
5/27/21    6.310765e+04
5/28/21    6.313185e+04
5/29/21    6.314522e+04
Name: California, Length: 499, dtype: float64

California had the most cases by 29th May 2021 with a sum of 63,145 cases. (6.3145*10^4)